{{cookiecutter.profile_name}}/slurm-sidecar.py

```diff
@@ -0,0 +1,312 @@
+        self.prev_call = 0.0
+        #: Time to sleep between iterations in seconds.  Thread can only be
+        #: terminated after this interval when waiting.
+        self.sleep_time = sleep_time
+        #: Maximal running time to accept for call to ``squeue``.
+        self.squeue_timeout = squeue_timeout
+        #: Maximal number of tries if call to ``squeue`` fails.
+        self.max_tries = max_tries
+        #: Dict mapping the job id to the job state string.
+        self.states = {}
+        #: Make at least one call to squeue, must not fail.
+        logger.debug("initializing trhead")
+        self._call_squeue(allow_failure=False)
+        self.prev_call = time.time()
+
+    def _work(self):
+        """Execute the thread's action"""
+        while not self.stopped.is_set():
+            now = time.time()
+            if now - self.prev_call > self.squeue_wait:
+                self._call_squeue()
+                self.prev_call = now
+            time.sleep(self.sleep_time)
+
+    def get_state(self, jobid):
+        """Return the job state for the given jobid."""
+        jobid = str(jobid)
+        if jobid not in self.states:
+            self.states[jobid] = self._get_state_sacct(jobid)
+        return self.states.get(jobid, "__not_seen_yet__")
+
+    def register_job(self, jobid):
+        """Register job with the given ID."""
+        self.states.setdefault(jobid, None)
+
+    def _get_state_sacct(self, jobid):
+        """Implement retrieving state via sacct for resuming jobs."""
+        cluster = CookieCutter.get_cluster_option()
+        cmd = ["sacct", "-P", "-b", "-j", jobid, "-n"]
+        if cluster:
+            cmd.append(cluster)
+        try_num = 0
+        while try_num < self.max_tries:
+            try_num += 1
+            try:
+                logger.debug("Calling %s (try %d)", cmd, try_num)
+                output = subprocess.check_output(cmd, timeout=self.squeue_timeout, text=True)
+                break
+            except subprocess.TimeoutExpired as e:
+                logger.debug("Call to %s timed out (try %d of %d)", cmd, try_num, self.max_tries)
+            except subprocess.CalledProcessError as e:
+                logger.debug("Call to %s failed (try %d of %d)", cmd, try_num, self.max_tries)
+        if try_num >= self.max_tries:
+            raise Exception("Problem with call to %s" % cmd)
+        else:
+            parsed = {x.split("|")[0]: x.split("|")[1] for x in output.strip().split("\n")}
+            logger.debug("Returning state of %s as %s", jobid, parsed[jobid])
+            return parsed[jobid]
+
+    def stop(self):
+        """Flag thread to stop execution"""
+        logger.debug("stopping thread")
+        self.stopped.set()
+
+    def _call_squeue(self, allow_failure=True):
+        """Run the call to ``squeue``"""
+        cluster = CookieCutter.get_cluster_option()
+        try_num = 0
+        cmd = [SQUEUE_CMD, "--me", "--format=%i,%T", "--state=all"]
```

Comment:
The '--me' option is supported by slurm releases newer than 20.02.(https://github.com/SchedMD/slurm/commit/c15b6c7f65a7a80093bfa526ffbe6deff9a0209f)  To make it backwards compatible with  older versions I suggest something like:
`
cmd = [SQUEUE_CMD, "--user", os.environ.get('USER')" , "--format=%i,%T", "--state=all"] 
`

Reply:
I agree with the --user option.

Reply:
Done.